addons/sale_project/models/sale_order.py

```diff
@@ -229,6 +229,8 @@ def _timesheet_create_project(self):
                 'sale_order_id': self.order_id,
             })
         else:
+            if len(self.order_id.order_line.filtered(lambda line: line.product_id.service_tracking == 'project_only')) == 1:
```

Comment:
Why not doing a `search_count` instead of doing many "prefetching"? 😅 
```suggestion
            project_only_sol_count = self.env['sale.order.line'].search_count([
                ('order_id', '=', self.order_id.id),
                ('product_id.service_tracking', '=', 'project_only'),
            ])
            if project_only_sol_count == 1:
```